src/nupic/algorithms/Cells4.cpp

```diff
@@ -188,16 +188,14 @@ bool Cells4::computeUpdate(UInt cellIdx, UInt segIdx, CStateIndexed& activeState
       highWaterSize = segment.size();
       newSynapses.reserve(highWaterSize);
     }
-    for (UInt i = 0; i != segment.size(); ++i)
+    for (UInt i=0; i<segment.size(); ++i) //FIXME why we skip segment[0] here?
```

Comment:
thanks, I learned something new today, thought the diff between ++i/i++ would affect the for-loop too. All makes sense now.